cleverhans/confidence_report.py

```diff
@@ -0,0 +1,129 @@
+"""Functionality for making confidence reports.
+
+A confidence report is a dictionary.
+Each dictionary key is the name of a type of data:
+  clean : Clean data
+  bundled : bundled adversarial examples
+Each value in the dictionary contains an array of bools indicating whether
+the model got each example correct and an array containing the confidence
+that the model assigned to each prediction.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import logging
+
+import numpy as np
+import tensorflow as tf
+
+from cleverhans.utils import set_log_level
+from cleverhans.serial import load
+
+# Defaults. Imported elsewhere so that command line script defaults match
+# function defaults.
+TRAIN_START = 0
+TRAIN_END = 60000
+TEST_START = 0
+TEST_END = 10000
+WHICH_SET = 'test'
+
+def make_confidence_report_bundled(filepath, train_start=TRAIN_START,
+                                   train_end=TRAIN_END, test_start=TEST_START,
+                                   test_end=TEST_END, which_set=WHICH_SET):
+  """
+  Load a saved model, gather its predictions, and save a confidence report.
```

Comment:
We would need that kind of check if we were writing a function where someone passes us a Model object reference and later we plan to call `save` on it. Here they pass us a filepath and we call `load` on the filepath. Because the model is already in a .joblib file we know it was picklable.

Reply:
Thanks for clarifying, I agree with you